src/Symfony/Component/EventDispatcher/Debug/TraceableEventDispatcherInterface.php

```diff
@@ -33,9 +34,4 @@ public function getCalledListeners();
      * @return array An array of not called listeners
      */
     public function getNotCalledListeners();
-
-    /**
-     * Resets the trace.
-     */
-    public function reset();
```

Comment:
I suppose we should change the `reset()` method in `EventDataCollector` to read `$this->dispatcher instanceof ResettableInterface` instead of `$this->dispatcher instanceof TraceableEventDispatcherInterface`.

Reply:
That being said, is this `reset()` method really tied with dependency injection? I think `Symfony\Contract\DependencyInjection\ResettableInterface` should be `Symfony\Contract\HttpKernel\ResettableInterface` instead, no?

Reply:
We could even use `Symfony\Contract\Service\ResettableInterface`, which is IMHO even better.

Reply:
`TraceableEventDispatcherInterface::reset()` means "Resets the trace" like description in phpdoc says.
But what does mean `TraceableEventDispatcher extends ResettableInterface` or just `EventDispatcher implements ResettableInterface `?
"Resets event dispatcher" (remove all subscribers?)?
"Resets trace"?
When `reset()` method is attached to specific concept/model, you have description of the method, it makes whole model more cohesive.
From my PoV, such approach ruins cohesion and it messes up semantics of `reset()` method for each context.

Just imagine one would notice `length()` method on `Line` and `Movie` classes. `Line::length()` returns meters (integer) and `Movie::length()` returns seconds (integer). Does it look like a good idea to make 

```php
interface HasLength
{
    public function length(): int;
}
```
? I don't think so.